tools/wire-format-reference.py

```
@@ -315,11 +324,30 @@ def _verify_vector(base: dict, bins: dict, msgpack) -> str:
 
+    # optional: liblz4 DECODE-only conformance — encode agreement reported, never
+    # asserted; see module docstring / spec 'Compressed-byte reproducibility' (LAB-1751).
+    lz4_note = ""
+    if lz4_block is not None:
+        inp = bytes.fromhex(base["input_hex"])
+        try:
+            got = lz4_block.decompress(data, uncompressed_size=size)
+        except (lz4_block.LZ4BlockError, OverflowError, MemoryError) as e:
+            # convert to the guarded type so a bad vector (corrupt stream, or an
+            # oversized size that liblz4 rejects/pre-allocates) fails itself, not the run
+            raise AssertionError(f"liblz4 rejects pinned compressed_data: {e}") from e
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)